luarules/gadgets/scav_spawner_defense.lua

```diff
@@ -968,12 +976,34 @@ if gadgetHandler:IsSyncedCode() then
 		return nil
 	end
 
+	local function calculateDifficultyMultiplier(peakScavPower, totalPlayerTeamPower)
+		-- Calculate the ratio
+		local ratio = peakScavPower / totalPlayerTeamPower
+	
+		-- Convert the ratio to a value between 0 and 1 for ratios between 1/5 and 1/3
+		if ratio >= 1/3 then
+			dynDifficulty0to1 = 0
+		elseif ratio <= 1/5 then
+			dynDifficulty0to1 = 1
+		else
+			dynDifficulty0to1 = (1/3 - ratio) / (1/3 - 1/5) --because 1/5 seems to be where players are doing well against scavs, while 1/3rd means they're losing.
+		end
+	
+		-- Calculate dynDifficultyClamped based on dynDifficulty0to1
+		dynDifficultyClamped = 0.85 + (dynDifficulty0to1 * (1.05 - 0.85)) --change 1.05 and .85 in these parenthesis to change the clamping range for the multiplier.
+	end
```

Comment:
These thresholds should be extracted to variables and given descriptive names, rather than magic values.
e.g.
```
local lowerScavPowerRatio = 1/5
local upperScavPowerRatio = 1/3
local minDynamicDifficulty = 0.85
local maxDynamicDifficulty = 1.05
```